pkg/controller/clusterdeployment/clusterprovisions.go

```diff
@@ -367,6 +368,38 @@ func setAzureResourceGroupFromMetadata(cd *hivev1.ClusterDeployment, cm *hivev1.
 	cm.Platform.Azure.ResourceGroupName = &rg
 }
 
+func setGCPNetworkProjectIDFromMetadata(cd *hivev1.ClusterDeployment, cm *hivev1.ClusterMetadata, pmjson []byte, logger log.FieldLogger) {
+	if pmjson == nil {
+		return
+	}
+	if cd.Spec.Platform.GCP == nil {
+		return
+	}
+
+	im := new(installertypes.ClusterMetadata)
+	if err := json.Unmarshal(pmjson, &im); err != nil {
+		logger.WithError(err).Error("Could not unmarshal ClusterMetadata!")
+		return
+	}
+
+	if im.GCP == nil {
+		logger.Warn("ClusterMetadata unexpectedly has no GCP section")
+		return
+	}
+	npid := im.GCP.NetworkProjectID
```

Comment:
Does this section require a check for whether `im.GCP.NetworkProjectID` is not nil, which would be different from an empty string? (Assuming it's not necessary but would like to know why)

Reply:
Is the field in installertypes.ClusterMetadata a pointer? I thought it was a string type.

Reply:
Oh, you're right. Makes sense.

Reply:
[ref](https://github.com/openshift/hive/blob/a9aaf72f8d1849aa5b5f2e3d6e137e93d9fbbdb5/vendor/github.com/openshift/installer/pkg/types/gcp/platform.go#L26)